### PR TITLE
#424: DNSPolicy basic definition

### LINKED ISSUE #424: How to define a basic DNSPolicy
**What**

add a section under the how-to sections of https://docs.kuadrant.io/multicluster-gateway-controller/ to explain the purpose of and how to define a basic DNSPolicy. It can then link off to or expand into more advanced docs and topics such as using GEO dns and Custom Weighting  plus health checks

### PR DESCRIPTION
**Pull request by sergioifg94**
## What

> Closes #424 

Add document describing how to define a basic DNSPolicy

* See document [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/90b8d2b815031bb1f35180f4268c68acbd3d8554/docs/how-to/dnspolicy_basic.md)

### CODE REVIEW

**Comment by jasonmadigan** [comment]
Think is good for a quick setup to take DNSPolicy for a spin, with https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/docs/dns-policy.md delves into more of the details with advanced usage. It sort of occurred to me that maybe the two could be merged - not sure.

**Comment by sergioifg94** [comment]
@jasonmadigan Yes I was struggling to come up with an approach for this that is not made redundant by the DNSPolicy reference, but I guess there might be some value in having a separate page just for the essential, and referencing the other one for advanced usage

File: docs/how-to/dnspolicy_basic.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Defining a basic DNSPolicy
+
+## What is a DNSPolicy
+
+DNSPolicy is a Custom Resource Definition supported by the MGC that follows the
```
Comment (maleck13) [inline docs/how-to/dnspolicy_basic.md:5]:
```suggestion
DNSPolicy is a Custom Resource Definition supported by the Multi-Cluster Gateway Controller (MGC) that follows the
```

File: docs/how-to/dnspolicy_basic.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[policy attachment model](https://gateway-api.sigs.k8s.io/references/policy-attachment/), which allows users to enable and configure DNS against the Gateway.
```
Comment (maleck13) [inline docs/how-to/dnspolicy_basic.md:6]:
```suggestion
[policy attachment model](https://gateway-api.sigs.k8s.io/references/policy-attachment/), which allows users to enable and configure DNS against the Gateway leveraging an existing cloud based DNS provider.
```

File: docs/how-to/dnspolicy_basic.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+This document describes how to enable DNS by creating a basic DNSPolicy
+
+## Pre-requisites
+
+* A [ManagedZone](managedZone.md) has been created
+* A Gateway has been created with an attached HTTPRoute
```
Comment (maleck13) [inline docs/how-to/dnspolicy_basic.md:13]: maybe we should call out that the DNSPolicy can be created before the HTTPRoute but that the record will only be created once a route is attached to the gateway?

File: docs/how-to/dnspolicy_basic.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+> See [the OCM walkthrough](ocm-control-plane-walkthrough.md) for step by step
+instructions on deploying these with a simple application.
+
+## Steps
+
+The DNSPolicy will target the existing Multi Cluster Gateway, resulting in the
+creation of DNS Records targeting the Gateway listeners, as well as optional
```
Comment (maleck13) [inline docs/how-to/dnspolicy_basic.md:21]:
```suggestion
creation of DNS Records for each of the Gateway listeners backed by a managed zone, as well as optional
```

File: docs/how-to/dnspolicy_basic.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+```sh
+kubectl get dnsrecords -A
+```
+
+The DNS records will be propagated in a few minutes, and the application will
+be available through the defined hosts.
+
+## Advanced DNS configuration
+
+The DNSPolicy supports other optional configuration options like load balancing
```
Comment (maleck13) [inline docs/how-to/dnspolicy_basic.md:51]:
```suggestion
The DNSPolicy supports other optional configuration options like geographic and weighted load balancing and health checks
```

**Review by maleck13 — COMMENTED** [review]
few suggestions but looking reasonable

**Comment by sergioifg94** [comment]
Thanks @maleck13, I just pushed the changes with your suggestions

**Review by maleck13 — APPROVED** [review]

**Pull request merged**